lib/generate/browserCollector.js

```diff
@@ -0,0 +1,191 @@
+const excludedModules = require('./excludedModules');
+
+/**
+ * Add a listener to the frame which will update the default requirejs config
+ * with a callback function. The callback function will be called immediately on
+ * requirejs being loaded, allowing us to attach a listener to onResourceLoad
+ * immediately and always capture everything loaded. By using "mutation" as the
+ * polling method, we can intercept the window.require object after it is output
+ * to the page and parsed by the browser, but before requirejs has loaded.
+ *
+ * The listener will build two global objects - tree and paths - tree being a
+ * list of all dependencies keyed by module and paths being a list of real paths
+ * for each defined name.
+ *
+ * After resources are loaded (defined as being 5 seconds after the final load
+ * event fires), set window global modulesLoaded to true so that the collectors
+ * can process the data.
+ *
+ * @param {import('puppeteer').Frame} frame
+ */
+const addListener = (frame) => {
+    return frame.waitForFunction(
+        (excludedModules) => {
+            if (typeof window.require === 'object') {
+                window.modulesLoadedTimer = null;
```

Comment:
This doesn't need to be defined at global scope, but works fine as-is. (It could be a local variable.)
```suggestion
                let modulesLoadedTimer;
```